explainer.md

```diff
@@ -147,17 +145,51 @@ function OnSessionStarted(session) {
   vrSession.depthNear = 0.1;
   vrSession.depthFar = 100.0;
 
-  // The content that will be shown on the device is defined by the session's
-  // baseLayer. In non-exclusive The baseLayer is not used for presentation, but
-  // the canvas dimensions are used to construct the projection matrices.
-  vrLayer = new VRCanvasLayer(vrSession, glCanvas);
-  vrSession.baseLayer = vrLayer;
+  // Ensure the canvas context is compatible and create the VRLayer.
+  SetupCanvasLayer().then(() => {
+    // Start the render loop
+    vrSession.commit().then(OnFirstVRFrame);
+  });
+}
+```
 
-  // Start the render loop
-  vrSession.commit().then(OnFirstVRFrame);
+### Setting up a VRLayer
+
+The content to present to the device is defined by a [`VRLayer`](https://w3c.github.io/webvr/#interface-vrlayer). In the initial version of the spec only one layer type, `VRCanvasLayer`, is defined and only one layer can be used at a time. This is set via the `VRSession.baseLayer` attribute. (`baseLayer` because future versions of the spec will likely enable multiple layers, at which point this would act like the `firstChild` attribute of a DOM element.)
+
+In order for a WebGL canvas to be used with a `VRCanvasLayer`, it's context must be _compatible_ with the `VRDevice`. This can mean different things for different environments. For example, on a desktop computer this means that the context should be created against the graphics adapter that the `VRDevice` is physically plugged into. On most mobile devices though, that's not a concern and so the context will always be compatible. In either case, the WebVR application will need to take steps to ensure WebGL context compatibility before using it with a `VRLayer`.
```

Comment:
on a desktop computer this means that the context should be created -> on a desktop computer this may mean the context must be created